rust/agama-cli/src/auth.rs

```diff
@@ -1,12 +1,57 @@
-use agama_lib::auth::AuthToken;
+use agama_lib::{auth::AuthToken, error::ServiceError};
 use clap::Subcommand;
 
 use crate::error::CliError;
+use agama_lib::base_http_client::BaseHTTPClient;
 use inquire::Password;
 use reqwest::header::{HeaderMap, HeaderValue, CONTENT_TYPE};
 use std::io::{self, IsTerminal};
 
-const DEFAULT_AUTH_URL: &str = "http://localhost/api/auth";
+/// HTTP Client for auth queries
+struct AuthHTTPClient {
+    api: BaseHTTPClient,
```

Comment:
I think this is wrong as BaseHTTPClient expects already authenticated entry. So auth queries should not set authentication. And also usage below does not use self.api...I expect that this case issue that @mvidner face.

Reply:
I've introduced new version, which doesn't look for authentication token.